src/Components/App/App.js

```diff
@@ -62,22 +68,74 @@ class App extends Component {
   }
 
   setActivities = (activities) => {
+    activities.forEach(activity => {
+      this.getActivityData(activity)
+    })
     this.setState({activities: [...activities]})
   }
 
   determineRandomActivity = () => {
+    let allFilteredActivities = this.state.activities.reduce((filtered, activity) => {
+      if (activity.includes('games')) {
+        activity = activity.replace(/ games/gi, 'Games')
+      }
+      let filteredActivities = this.filterActivity(activity, this.state[`${activity}Answers`])
+      filtered.push(filteredActivities)
+      return filtered
+    }, [])
+    let final = allFilteredActivities.flat()
+    let randomNumber = Math.floor(Math.random() *final.length)
+    return final[randomNumber]
+  }
 
-    // redirect to temporary loading page while the below logic is run
-    // what activities did the user select?
-    // fetch all activities and place somewhere to be filtered through, here? Result component?
-    // if Result component we will also have to pass the answers of the questions down
-    // filter through All results from fetch with specific conditions based on selected answers
-    // store these further filtered results in somewhere where we can access them more than once, state?
-    // randomly choose one of these further filtered activities, maybe remove this from state at that time? 
-    // render component that will display the previewcard
-    // let user select to see more info or skip
-    // if user selects skip, get another random activity from set which should no longer include the one they skipped if we removed it
-    // Should user be able to skip once they have selected to see more info? If they are no longer interested?
+  filterActivity = (activity, answers) => {
+    //refactor this - break out into helper functions
+    let genreGroup = ['movies', 'podcasts', 'music']
+    let gamesGroup = ['boardGames', 'cardGames']
+    let ageGroup = ['movies', 'music']
+    let possibleSuggestions;
+    if (genreGroup.includes(activity)) {
+      possibleSuggestions = this.state[activity].filter(singleActivity => {
+        if (activity === 'movies') {
+          return singleActivity.genre.some(genre => answers.includes(genre))
+        }
+        return answers.some(genre => genre === singleActivity.genre)
+      })
```

Comment:
While it's not the prettiest function, It has some pretty solid logic regarding the filtering of the activities based on user responses.

I think we can definitely split these up separate functions that are only ever responsible for the specific activity type